components/ai_chat/resources/page/components/conversation_list/index.tsx

```diff
@@ -77,14 +76,23 @@ function ConversationList(props: ConversationListProps) {
     props.onLastElementHeightChange()
   }, [conversationHistory.length, lastEntryElementRef.current?.clientHeight])
 
+  const lastAssistantId = React.useMemo(() => {
+    for (let i = conversationHistory.length - 1; i >= 0; i--) {
+      if (conversationHistory[i].characterType === mojom.CharacterType.ASSISTANT) {
+        return i
+      }
+    }
```

Comment:
although i'm a fan of for loops and understand why you're not using Array.reverse().forEach iterator here, i wonder if `[...conversationHistory].reverse().forEach` is more readable. however, it duplicates the array. not sure if duplication outweighs the readability.

Reply:
Yeah avoiding the duplication on every conversation history change (which is very frequent especially during generation when it can be modified character-by-character). I can add a comment to make it clearer?

Reply:
yes, a comment would be good